crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -180,18 +180,40 @@ impl SimpleFaucet {
 
         match gas_coin_response {
             GasCoinResponse::ValidGasCoin(coin_id) => {
-                let result = self
-                    .execute_pay_sui_txn_with_retrials(
+                let response = match timeout(
```

Comment:
`map_err` might be helpful here, though I'm not sure how it interacts with `warn!`?

Reply:
I think it would Interact fine with map_err, but I don't know that it gains you much here.

Reply:
not much, other than like one level of indentation. Amazing, I know